coding-concepts.js

```diff
@@ -9,46 +9,46 @@
 const colors = ["tangerine", "magenta", "lilac", "daffodil"]
 // console.log(colors.push("indigo"))
 
-// a) Your answer:
-// b) Verify and explain:
+// a) Your answer: 5 
+// b) Verify and explain: indigo would be the 5 added element being pushed onto the funcion
 
 // --------------------1) What will this log?
 
 const cohort = "LEARN 2023"
 // console.log(cohort.length)
 
-// a) Your answer:
-// b) Verify and explain:
+// a) Your answer: 10
+// b) Verify and explain: their are 10 symbols in the function including the space
 
 // --------------------2) What will this log?
 
 const greeting = "Hello World!"
 // console.log(greeting[4])
 
-// a) Your answer:
-// b) Verify and explain:
+// a) Your answer: 0
+// b) Verify and explain: the 4 is in brackets which i belive makes it n array? so it indicates their is no 4 in the the greeting?
 
 // --------------------3) What will this log?
 
 const languages = ["JavaScript", "Ruby", "Python", "C++"]
 const index = 1
 // console.log(languages[index])
 
-// a) Your answer:
-// b) Verify and explain:
+// a) Your answer: ruby
+// b) Verify and explain: the function is an array so within the brackets the index starts at 0 therefore ruby is 1
 
 // --------------------4) What will this log?
 
 const weekendDays = ["saturday", "sunday"]
 // console.log(weekendDays.toUpperCase())
 
-// a) Your answer:
-// b) Verify and explain:
+// a) Your answer: i got an error
+// b) Verify and explain: no function equals error
 
 // --------------------5) What will this log?
 
 const dataTypes = ["number", "string", "Boolean", "undefined"]
-// console.log(typeof dataTypes.length)
+ console.log(typeof dataTypes.length)
 
-// a) Your answer:
-// b) Verify and explain:
+// a) Your answer: number 
+// b) Verify and explain: im not sure why "number" popped up
```

Comment:
The `typeof` will return the data type of whatever comes right after it. Since this example has `dataTypes.length`, which returns the length of the array, the typeof is looking at that length, and sees that it is a number.